deltacat/compute/stats/utils/intervals.py

```diff
@@ -1,62 +1,88 @@
-from typing import Set, Tuple
-from collections import Iterable
+from typing import Set, Tuple, List, Iterable, Union
 
+UnboundedRange = Union[int, None]
+NumericalRange = Union[int, float]
```

Comment:
Minor: Maybe add a comment stating that float is only added to support math.inf?

Reply:
On second thought, I think it would make this code and subsequent code much more clear to just drop `math.inf` support which allows us to remove `float` from the signature here.

Then your whole typing here is also reduced down to just one type:
`DeltaRange = Optional[int]`

Reply:
I think we would still need the `float` to account for the numerical representation of None, to do the actual sort/merge. But I agree that the input and output signature of `merge_intervals` can be reduced down to `DeltaRange = Optional[int]`